### ISSUE TITLE
Registering ignores capitalization

### ISSUE THREAD

**Issue by krithel** [atosorigin/PAMM_Skeleton#4]
Currently, when registering, all captialization is ignored in the back-end, and the username is saved in lowercase. This should be changed to save the input username as-is, but when checking for duplicates or authenticating a user, case should be ignored.

**Comment by tim-m-robinson** [comment]
we might want to think about this.. for example, what if we whitelist against gmail or other external accounts where John.Smith@mail.com is distinct from john.smith@mail.com.

**Comment by krithel** [comment]
Interesting, I took it for granted that emails were case-insensitive since everyone treats them as such, but you are right, that assumption was wrong. Currently we don't have email entry, but that would need to be taken into account.

The thought behind pseudo-case insensitive usernames is entirely for user convenience. If a user registers as 'Johnsmith1', it seems reasonable to allow them to log in under 'johnsmith1'. If this were to be extended to have email as the key identifier, that would clearly have to change, but I am not sure about more generic usernames.

**Comment by tim-m-robinson** [comment]
This is where the original draft specs (called RFCs) for the Interwebs are very useful. 
http://stackoverflow.com/questions/9807909/are-email-addresses-case-sensitive

For this case though it is _consistency_ that is the overriding factor. If we define that user names are case insensitive then we should have:
1) A text entry field that only accepts lowercase
2) A web service that reformats input to lowercase
3) user name lookup that compares lowercase
4) data persistence that stores in lowercase

**Comment by ghost** [comment]
Yes, Tim is right, RFC 5231 does in fact state that text to the left of the "@" sign is case sensitive. However [gmail does not discriminate](http://www.computerforum.com/threads/are-gmail-addresses-case-sensitive.228339/). 

This needs some sort of testing. Something like [mailosaur](https://mailosaur.com/).

**Comment by ghost** [comment]
Oops, didn't mean to close it. Sorry.
